@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -193,14 +193,26 @@ export class TileDataSource<TileType extends Tile = Tile> extends DataSource {
         await Promise.all([this.m_options.dataProvider.register(this), this.m_decoder.connect()]);
         this.m_isReady = true;
 
-        this.m_decoder.configure(undefined, undefined, undefined, {
+        this.m_decoder.configure(undefined, {
             storageLevelOffset: this.m_options.storageLevelOffset
         });
     }
 
     /** @override */
-    setStyleSet(styleSet?: StyleSet, definitions?: Definitions, languages?: string[]): void {
-        this.m_decoder.configure(styleSet, definitions, languages);
+    setStyleSet(
+        options: StyleSetOptions | StyleSet,
+        definitions?: Definitions,
+        languages?: string[]
+    ): void {
+        if (!options?.hasOwnProperty("styleSet")) {
+            this.m_decoder.configure({
+                styleSet: options as StyleSet,
+                definitions,
+                languages
+            });
+        } else {
+            this.m_decoder.configure(options as StyleSetOptions);
+        }
```

Comment:
we could add the two prototype restrictions below... or we could add a runtime warning message if a user pass both options and definitions/languages
```
setStyleSet(options: StyleSetOptions);
setStyleSet(styleSet: StyleSet, definitions?: Definitions, languages?: string[]);
```